cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -1052,7 +1052,27 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size_t, 8)
   // Exit super early for simple types if the page does not need to be decoded
   if constexpr (not has_lists_t and not has_strings_t and not has_nesting_t) {
     if (not page_mask[page_idx]) {
-      pp->num_nulls  = pp->num_rows;
+      auto const page_num_rows = [=]() {
```

Comment:
Isn't this whole block redundant to what we're already doing in `setup_local_page_info`?

https://github.com/rapidsai/cudf/blob/df567254b81fc8d8c6d8c4f636d1c288717b77c2/cpp/src/io/parquet/page_decode.cuh#L1221

Reply:
On the other hand we can just get rid of this whole if block altogether if needed.

Reply:
It's still duplicated code though. Kind of complex logic too (these kinds of row and page boundary checks are one of the most common causes of bugs).  Suggest at least refactoring into a common function.